src/LibDDRV.sol

```diff
@@ -103,16 +107,10 @@ library LibDDRV {
         // Setup an empty level
         forest.levels[level].weight = 0;
         forest.levels[level].roots = 0;
+
         // Setup an empty queue
-        // Qₗ₊₁ = ∅
-        assembly {
-            // Set the queue to the free pointer
-            ptr1 := mload(fp)
-            head1 := add(ptr, word)
-            // One word is reserved here to act as a header for the queue,
-            // to check if a range is already in the queue.
-            tail1 := head
-        }
+        (ptr1, head1, tail1) = new_queue(head);
```

Comment:
The new_queue function shouldn't care about the old head, can just use fp = 0x40 to get the next free memory pointer